### PR TITLE
Add support for image params for disconnected

### LINKED ISSUE #293: Explore handling of Images for individual components
As we are moving towards opinionated deployments, explore if we want to expose image fields through `vars`.

Note `vars` is also deprecated field in kustomize

### PR DESCRIPTION
**Pull request by VaishnaviHire**
Fixes #293 
## How Has This Been Tested?
1. Deploy test catalogsource
```
quay.io/vhire/opendatahub-operator-catalog:v2.0.1
```
2. Enable datascience pipelines component
3. Verify Datascience pipelines operator pod defines env variables added in the CSV file.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: pkg/deploy/deploy.go [hunk 1]
```diff
@@ -224,4 +225,74 @@ func manageResource(owner metav1.Object, ctx context.Context, cli client.Client,
...
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return err
+	}
+
+	// Update images with env variables
+	for key, _ := range envMap {
+		relatedImageValue := os.Getenv(imageParamsMap[key])
```
Comment (zdtsw) [inline pkg/deploy/deploy.go:258]: why need use `os.Getenv `here?

**Review by zdtsw — APPROVED** [review]
/lgtm

**Review by etirelli — APPROVED** [review]
LGTM

**Pull request merged**